server/src/main/java/org/elasticsearch/action/admin/cluster/stats/TransportClusterStatsAction.java

```diff
@@ -94,32 +80,8 @@ protected ClusterStatsNodeResponse newNodeResponse() {
     protected ClusterStatsNodeResponse nodeOperation(ClusterStatsNodeRequest nodeRequest) {
         NodeInfo nodeInfo = nodeService.info(true, true, false, true, false, true, false, true, false, false);
         NodeStats nodeStats = nodeService.stats(CommonStatsFlags.NONE,
-                true, true, true, false, true, false, false, false, false, false, false, false);
-        List<ShardStats> shardsStats = new ArrayList<>();
-        for (IndexService indexService : indicesService) {
-            for (IndexShard indexShard : indexService) {
-                if (indexShard.routingEntry() != null && indexShard.routingEntry().active()) {
-                    // only report on fully started shards
-                    CommitStats commitStats;
-                    SeqNoStats seqNoStats;
-                    try {
-                        commitStats = indexShard.commitStats();
-                        seqNoStats = indexShard.seqNoStats();
-                    } catch (AlreadyClosedException e) {
-                        // shard is closed - no stats is fine
-                        commitStats = null;
-                        seqNoStats = null;
-                    }
-                    shardsStats.add(
-                        new ShardStats(
-                            indexShard.routingEntry(),
-                            indexShard.shardPath(),
-                            new CommonStats(indicesService.getIndicesQueryCache(), indexShard, SHARD_STATS_FLAGS),
-                            commitStats,
-                            seqNoStats));
-                }
-            }
-        }
+                true, true, true, false, true, false, false, false, false, false, false, false, true);
```

Comment:
Are we sure that we want to always return shard stats in the cluster stats?

Reply:
The old code was always doing the equivalent lookup of shard stats without any conditions to block it, so I think it's the right thing to do.

Reply:
`TransportClusterStatsAction` - it's the loop that this is replacing.